plugin/kubernetes/kubernetes.go

```diff
@@ -110,21 +117,30 @@ func (k *Kubernetes) Services(state request.Request, exact bool, opt plugin.Opti
 
 	case dns.TypeNS:
 		// We can only get here if the qname equals the zone, see ServeDNS in handler.go.
-		ns := k.nsAddr()
-		svc := msg.Service{Host: ns.A.String(), Key: msg.Path(state.QName(), "coredns")}
+		ns := k.nsAddr(external)
+		svc := msg.Service{Host: ns[0].A.String(), Key: msg.Path(state.QName(), "coredns")}
 		return []msg.Service{svc}, nil
 	}
 
 	if state.QType() == dns.TypeA && isDefaultNS(state.Name(), state.Zone) {
-		// If this is an A request for "ns.dns", respond with a "fake" record for coredns.
-		// SOA records always use this hardcoded name
-		ns := k.nsAddr()
-		svc := msg.Service{Host: ns.A.String(), Key: msg.Path(state.QName(), "coredns")}
-		return []msg.Service{svc}, nil
+		// create an A record for the NS target
+		nss := k.nsAddr(external)
+		var svcs []msg.Service
+		for _, ns := range nss {
+			if ns.A.String() == "0.0.0.0" {
```

Comment:
what?

Reply:
Yeah - I should have commented the code here. I am using `0.0.0.0` to mean a null IP (i.e. we don't know the IP). I will refactor the nsAddr() function to make this less ugly.

Reply:
I added comments, and use nil instead, which is a lot clearer.